File: 해시 - 완주하지 못한 선수/소스.cpp
#include <stdio.h>
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <math.h>
#include <algorithm>
#include <string>
#include <sstream>
#include <algorithm>
#include <queue>
#include <math.h>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());

    string answer = participant.back();

    for (int i = 0; i < completion.size(); i++) {
        if (participant[i] != completion[i]) {
            answer = participant[i];
            break;
        }
    }

    return answer;
}

int main() {
    vector<string> participant = { "mislav", "stanko", "mislav", "ana" };
    vector<string> completion = { "stanko", "ana", "mislav" };
    string answer = solution(participant, completion);
    cout << "완주하지 못한사람 : " << answer;
    return 0;
}

File: 해시 - 완주하지 못한 선수/소스.js
function solution(participant, completion) {
  participant.sort();
  completion.sort();
  let answer = participant[participant.length - 1];
  for (const [i, v] of completion.entries())
    if (v != participant[i]) {
      answer = participant[i];
      break;
    }
  return answer;
}

(function () {
  const participant = ["mislav", "stanko", "mislav", "ana"];
  const completion = ["stanko", "ana", "mislav"];
  const sol = solution(participant, completion);
  console.log(sol);
})();
